ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/GenesisState.java

```diff
@@ -114,8 +114,13 @@ public static GenesisState fromConfig(
       final DataStorageConfiguration dataStorageConfiguration,
       final GenesisConfigFile genesisConfigFile,
       final ProtocolSchedule protocolSchedule) {
-    final var genesisStateRoot =
-        calculateGenesisStateRoot(dataStorageConfiguration, genesisConfigFile);
+    // for optimism mainnet, it will modify genesis state root.
+    final Hash genesisStateRoot;
+    if (!genesisConfigFile.getStateHash().isEmpty()) {
+      genesisStateRoot = Hash.fromHexStringLenient(genesisConfigFile.getStateHash());
```

Comment:
I don't understand why we cannot find the good state root hash with calculateGenesisStateRoot. seems to be an issue in the genesis or something else

Reply:
is this related to the massive genesis ?

Reply:
I understand know, we checked with Gary

Reply:
Would you move the stateHash override functionality into an OptimismGenesisState subclass?  

This will keep the optimisim specific behaviors in an easy to find class, so we can later create the necessary plugin-api interfaces to make op-besu a plugin.